account_reconcile_restrict_partner_mismatch/models/account_move_line.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.exceptions import UserError
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @property
+    def _check_partner_mismatch_on_reconcile(self):
+        """
+        Returns True if the partner mismatch check on reconcile should be done
+        """
+        self.ensure_one()
+        return bool(
+            self.company_id.restrict_partner_mismatch_on_reconcile
+            and not self.journal_id.no_restrict_partner_mismatch_on_reconcile
+            and self.account_id.account_type
+            in ("asset_receivable", "liability_payable")
+        )
+
+    def _reconcile_plan(self, reconciliation_plan):
+        # to be consistent with parent method
+        if self:
+            partners = set()
+            for line in self:
```

Comment:
`_reconcile_plan` is an `api.model` function, you should loop over the `reconciliation_plan` var instead (cf docstring to know how it must be structured)

Reply:
improved test-case and loop over reconciliation_plan as suggested